src/osd/PG.cc

```diff
@@ -4621,10 +4656,23 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
         assert(last_update_applied >= scrubber.subset_last_update);
 
         // build my own scrub map
-        ret = build_scrub_map_chunk(scrubber.primary_scrubmap,
-                                    scrubber.start, scrubber.end,
-                                    scrubber.deep, scrubber.seed,
-				    handle);
+	{
+	  epoch_t epoch = get_osdmap()->get_epoch();
+	  if (scrubber.preempt_left > 0) {
+	    unlock();
+	  }
+	  ret = build_scrub_map_chunk(scrubber.primary_scrubmap,
```

Comment:
build_scrub_map_chunk() uses the pg_backend that's destroyed when the pg is removed - do we need to include a way to wait for this to finish in PrimaryLogPG::on_{removal,shutdown}?

Reply:
yeah, probably.  we should probably do the same on an interval change too.